File: leetcode/easy/calculate-amount-paid-in-taxes.ts
//
// https://leetcode.com/problems/calculate-amount-paid-in-taxes/

// MINE
function calculateTax(brackets: number[][], income: number): number {
  let ans = 0,
    prev = 0
  for (let [upper, percent] of brackets) {
    if (upper <= income) {
      ans += ((upper - prev) * percent) / 100
      prev = upper
    } else {
      ans += ((income - prev) * percent) / 100
      return ans
    }
  }
  return ans
}

// ANSWER

File: leetcode/easy/calculate-amount-paid-in-taxes.py
# https://leetcode.com/problems/calculate-amount-paid-in-taxes/

class Solution:
    def calculateTax(self, brackets: List[List[int]], income: int) -> float:
        ans = prev = 0
        for upper, percent in brackets:
            if upper <= income:
                ans += (upper - prev) * percent / 100
                prev = upper
            else:
                ans += (income - prev) * percent / 100
                return ans
        return ans
